newIDE/app/src/ObjectEditor/Editors/SpriteEditor/SpritesList.js

```diff
@@ -1,160 +1,239 @@
+const ipcRenderer = electron ? electron.ipcRenderer : null;
+const path = optionalRequire('path');
+
+const gd = global.gd;
+
+const SPRITE_SIZE = 100; //TODO: Factor with Thumbnail
+
+var editedAnimationProp = null; // need this for piskel edits
+
+const styles = {
+  spritesList: {
+    whiteSpace: 'nowrap',
+    overflowY: 'hidden',
+  },
+  thumbnailExtraStyle: {
+    marginRight: 10,
+  },
+  addSpriteButton: {
+    ...thumbnailContainerStyle,
+    background: '#FFF',
+  },
+  spriteThumbnailImage: {
+    maxWidth: SPRITE_SIZE,
+    maxHeight: SPRITE_SIZE,
+    verticalAlign: 'middle',
+  },
+};
+
+const AddSpriteButton = SortableElement(({ displayHint, onAdd, onEdit }) => {
+  return (
+    <div style={styles.addSpriteButton}>
+      <IconButton onClick={onAdd} style={styles.spriteThumbnailImage}>
+        <Add />
+      </IconButton>
+      <IconButton onClick={onEdit} style={styles.spriteThumbnailImage}>
+        <Brush />
+      </IconButton>
+    </div>
+  );
+});
+
+const SortableSpriteThumbnail = SortableElement(
+  ({ sprite, project, resourcesLoader, selected, onSelect, onContextMenu }) => {
+    return (
+      <ImageThumbnail
+        selectable
+        selected={selected}
+        onSelect={onSelect}
+        onContextMenu={onContextMenu}
+        resourceName={sprite.getImageName()}
+        resourcesLoader={resourcesLoader}
+        project={project}
+        style={styles.thumbnailExtraStyle}
+      />
+    );
+  }
+);
+
+const SortableList = SortableContainer(
+  ({
+    direction,
+    project,
+    resourcesLoader,
+    onAddSprite,
+    onEditSprites,
+    selectedSprites,
+    onSelectSprite,
+    onSpriteContextMenu,
+  }) => {
+    const spritesCount = direction.getSpritesCount();
+    return (
+      <div style={styles.spritesList}>
+        {[
+          ...mapFor(0, spritesCount, i => {
+            const sprite = direction.getSprite(i);
+            return (
+              <SortableSpriteThumbnail
+                sprite={sprite}
+                key={sprite.ptr}
+                index={i}
+                selected={!!selectedSprites[sprite.ptr]}
+                onContextMenu={(x, y) => onSpriteContextMenu(x, y, sprite)}
+                onSelect={selected => onSelectSprite(sprite, selected)}
+                resourcesLoader={resourcesLoader}
+                project={project}
+              />
+            );
+          }),
+          <AddSpriteButton
+            displayHint={!direction.getSpritesCount()}
+            key="add-sprite-button"
+            disabled
+            index={spritesCount}
+            onAdd={onAddSprite}
+            onEdit={onEditSprites}
+          />,
+        ]}
+      </div>
+    );
+  }
+);
+
+export default class SpritesList extends Component {
+  componentDidMount() {
+    if(!ipcRenderer){return};
+    ipcRenderer.on('piskelSavedChanges', (event, piskelFramePaths) => {
+      const { direction, project } = editedAnimationProp.props;

```

Comment:
Can't we use `this.props` and completely remove `editedAnimationProp`?
Which by the way has not a proper naming as it's called `editedAnimationProp` but you assign it to `this`, which is a component, not to a list of props :) Anyway, pretty sure we can remove this, and as it's a "global" variables it's dangerous.

Reply:
I actually need to store it in a variable when opening piskel.  
This is how this function knows to which animation prop to save the changes to. If I use 'this'- all clips will get overwriten by piskel's edits. 
Another approach would be to pass it with the piskelData array. Can you suggest an alternative approach? :)

Reply:
EDIT:  what happens if I use this..
![gd-piskel-this](https://user-images.githubusercontent.com/6495061/39941640-1b2959f0-5555-11e8-8050-09563891b67f.gif)

Reply:
Mmm, not sure what is happening but you should not have to keep a reference to the component. Can you try to debug to see why it's adding a new animation?

Reply:
Will investigate this :)

Reply:
I noticed that it is running the actions inside the signal on each animation component.
I am assuming that electron's main function is sending the signal with the array to all animation components- triggering the actions on all.

Reply:
Another idea is to attach a boolean

edited: true

to the animation component edited and switch it back to false on the update :)
That would get rid of the global variable!

Reply:
I think we can save in a member variable of the class, the animation id and direction id being edited.
Otherwise I'll take a look by myself but I'm 100% sure we can do it cleanly without using a global variables which could cause hard to debug issues. All the software is written in React components using states and member variables in classes, no reason we can't do it there :)

Reply:
The boolean is a member of SpriteList.js

I have no idea how to get the animation ID and then later on use it to point to the correct animation  :(